### PR TITLE
Validate snapshot before inplace restore

### PR DESCRIPTION
**Pull request by ram-infrac**
**What type of PR is this?**
>bug

**What this PR does / why we need it**:
Add check to validate whether given snapshot is complete before performing
 in place restore

### CODE REVIEW

File: pkg/snapshot/controllers/snapshotrestore.go [hunk 1]
```diff
@@ -155,6 +157,12 @@ func (c *SnapshotRestoreController) handleInitial(snapRestore *stork_api.VolumeS
 			return fmt.Errorf("unable to get get snapshot  details %s: %v",
 				snapName, err)
 		}
+		if err := k8s.Instance().ValidateSnapshot(snapName,
```
Comment (disrani-px) [inline pkg/snapshot/controllers/snapshotrestore.go]: We shouldn't retry here. Just check if the snapshot is bound. If not raise an event and return nil, it'll be retried in the next reconcile loop

Comment (ram-infrac) [reply pkg/snapshot/controllers/snapshotrestore.go]: bound meaning whether spec.SnapshotDataName is not empty ?

Comment (disrani-px) [reply pkg/snapshot/controllers/snapshotrestore.go]: Set the 3rd param for `ValidateSnaphshot()` to false to not retry

Comment (ram-infrac) [reply pkg/snapshot/controllers/snapshotrestore.go]: set `retry` to false

**Pull request merged**